File: java/1022.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int sumRootToLeaf(TreeNode root) {
        return sum(root,0);
    }
    
    public int sum(TreeNode root,int cur){
        int sum = 0;
        if(root == null){
            return 0;
        }
        cur = cur * 2 + root.val;
        if(root.left == null && root.right == null){
            return cur;
        }
        if(root.left != null){
            sum = sum + sum(root.left,cur);
        }
        if(root.right != null){
            sum = sum + sum(root.right,cur);
        }
        return sum;
    }
}

File: go/1022.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func sumRootToLeaf(root *TreeNode) int {
	return sumRoot(root, 0)
}

func sumRoot(root *TreeNode, cur int) int {
	sum := 0
	if root == nil {
		return 0
	}
	cur = cur*2 + root.Val
	if root.Left == nil && root.Right == nil {
		return cur
	}
	if root.Left != nil {
		sum += sumRoot(root.Left, cur)
	}
	if root.Right != nil {
		sum += sumRoot(root.Right, cur)
	}
	return sum

}